presto-main/src/test/java/com/facebook/presto/security/TestAccessControlManager.java

```diff
@@ -114,6 +116,33 @@ public void testReadOnlySystemAccessControl()
         }
     }
 
+    @Test
+    public void testUsernamePrincipalMatchingAccessControl()
+    {
+        AccessControlManager accessControlManager = new AccessControlManager(createTestTransactionManager());
+        accessControlManager.setSystemAccessControl(UsernamePrincipalMatchingAccessControl.NAME, ImmutableMap.of());
+
+        accessControlManager.checkCanSetUser(null, USER_NAME);
+        accessControlManager.checkCanSetUser(new TestingPrincipal(USER_NAME), USER_NAME);
+        try {
+            accessControlManager.checkCanSetUser(PRINCIPAL, USER_NAME);
```

Comment:
you need to add `throw AssertionError("expected AccessDeniedExeption")` after this line. 
Otherwise you are not testing if `checkCanSetUser` actually thrown.